File: LeetCode/238. Product of Array Except Self/238. Product of Array Except Self.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int prod = 1, flag = 0, x = 0;
        for(int i = 0; i < nums.size(); i++){
            if(nums[i] != 0){
                prod *= nums[i];
            }
            else{
                flag++;
            }
        }
        vector <int> ans;
        for(int i = 0; i < nums.size(); i++){
            if(flag == 0){
                ans.push_back(prod/nums[i]);
            }
            else if(nums[i] != 0 && flag == 1){
                ans.push_back(0);
            }
            else if(flag > 1){
                ans.push_back(0);
            }
            else{
                ans.push_back(prod);
            }
        }
        return ans;
    }
    
};

File: LeetCode/238. Product of Array Except Self/238. Product of Array Except Self.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int[] answer = new int [nums.length];
        int flag = 0, multiply = 1;
        for(int i = 0; i < nums.length; i++){
            if(nums[i] == 0) { 
                flag = flag + 1;
                continue;
            }
            multiply = multiply * nums[i];
        }
        for(int i = 0; i < nums.length; i++) {
            if(flag > 1) {
                answer[i] = 0;
            } 
            else if(flag == 1 && nums[i] != 0) {
                answer[i] = 0;
            }
            else if(flag == 1 && nums[i] == 0) {
                answer[i] = multiply;
            }
            else
                answer[i] = multiply / nums[i];
        }
        return answer;
    }
}